opentelemetry-sdk/src/metrics/internal/sum.rs

```diff
@@ -1,22 +1,35 @@
 use std::sync::atomic::{AtomicBool, Ordering};
 use std::{
-    collections::{hash_map::Entry, HashMap},
-    sync::Mutex,
+    sync::{Arc, Mutex},
     time::SystemTime,
 };
 
 use crate::attributes::AttributeSet;
 use crate::metrics::data::{self, Aggregation, DataPoint, Temporality};
 use opentelemetry::{global, metrics::MetricsError};
+use std::hash::{Hash, Hasher};
+
+#[cfg(feature = "use_hashbrown")]
+use ahash::AHasher;
+#[cfg(feature = "use_hashbrown")]
+use hashbrown::{hash_map::Entry, HashMap};
+
+#[cfg(not(feature = "use_hashbrown"))]
+use std::collections::{
+    hash_map::{DefaultHasher, Entry},
+    HashMap,
+};
 
 use super::{
     aggregate::{is_under_cardinality_limit, STREAM_OVERFLOW_ATTRIBUTE_SET},
     AtomicTracker, Number,
 };
 
+type BucketValue<T> = Mutex<Option<HashMap<AttributeSet, T>>>;
+type Buckets<T> = Arc<[BucketValue<T>; 256]>;
```

Comment:
why 256? Is that something that should be done based on processor count?

Reply:
These are not designed as buckets for each thread/core. The selection of a fixed size of 256 buckets was based on confining the hash value to a `u8` size, and given a default cardinality limit of 2000, with a uniform distribution, each bucket would contain approximately 10 metric points. This can be made configurable if we want the application to define the optimal size of buckets based on their attribute distribution.